sim/src/main.rs

```diff
@@ -50,40 +50,73 @@ fn main() {
     }
 
     // Create simulator object.
-    let mut simulator: Simulator = Simulator::new(seed);
+    let storage_name = "storage";
+    let mut simulator = sim::simulator::Simulator::new(1); // always run with the seed 0 while we are checking tests
 
-    // node arguments go:  id, capacity, egress_rate, generation_rate, plugin
-    simulator.add_node("traffic-gen", 1, 1, 1, plugin_str);
-    simulator.add_node("node-1", 10, 1, 0, plugin_str);
-    simulator.add_node("node-2", 10, 1, 0, plugin_str);
-    simulator.add_node("node-3", 10, 1, 0, plugin_str);
-    simulator.add_node("node-4", 1, 1, 0, plugin_str); // in setting egress rate to 0, we are making a sink
-    simulator.add_node("node-5", 1, 1, 0, plugin_str); // in setting egress rate to 0, we are making a sink
-    simulator.add_storage("storage");
+    let regular_nodes = [
+        "frontend-v1",
+        "cartservice-v1",
+        "productcatalogservice-v1",
+        "currencyservice-v1",
+        "paymentservice-v1",
+        "shippingservice-v1",
+        "emailservice-v1",
+        "checkoutservice-v1",
+        "recomendationservice-v1",
+        "adservice-v1",
+    ]
+    .to_vec();
+    for node in &regular_nodes {
+        // node: ID, capacity, egress rate, generation rate, plugin
+        simulator.add_node(node, 10, 5, 0, plugin_str);
+    }
+    simulator.add_node("loadgenerator-v1", 10, 1, 1, None);
+    simulator.add_node("sink", 10, 1, 0, None); // rpc sink
+    simulator.add_storage(storage_name);
+
+    // add all connections to storage and to the sink - we want traces to be able to end arbitrarily
```

Comment:
How can a trace end arbitrarily?

Reply:
There is a sink that is connected to everything else.  Because it is everyone's neighbor, eventually the RPC will randomly go to the sink, where it disappears.

Reply:
I think sink should be an implicit thing. For example, in the [bookinfo](https://istio.io/latest/docs/examples/bookinfo/) app ratings and details should be sinks.

Reply:
Do you think we should make it an option when creating a node?

Reply:
Right now a sink just has zero output right? So an option does not seem necessary. Once we get to the point where nodes respond on requests we may have to rethink this.

Reply:
So do you still think we should have an implicit sink?  There is no sense of the "end" of a graph because an RPC can be sent back to mimic a response, so we'd implicitly be adding it to all nodes in the simulator.

Reply:
Yes, until we have implemented responses these should be implicit sinks. That gets close to a request graph at least. We will have to develop custom node logic soon to model the full productpage application anyway.

Reply:
Okay, it is created implicitly now.